react/src/lib/components/DeckGLMap/layers/map/mapLayer.ts

```diff
@@ -0,0 +1,351 @@
+                        ? propertyValue
+                        : valueRangeMin;
+                valueRangeMax =
+                    propertyValue > valueRangeMax
+                        ? propertyValue
+                        : valueRangeMax;
+            }
+        }
+    } else {
+        valueRangeMin = valueRange[0];
+        valueRangeMax = valueRange[1];
+    }
+
+    let colorMapRangeMin = colorMapRange?.[0] ?? valueRangeMin;
+    let colorMapRangeMax = colorMapRange?.[1] ?? valueRangeMax;
+    if (colorMapRangeMax < colorMapRangeMin) {
+        const temp = colorMapRangeMax;
+        colorMapRangeMax = colorMapRangeMin;
+        colorMapRangeMin = temp;
+    }
+
+    // Apply color map.
+    for (let i = 0; i < data.length; i += 4) {
+        const propertyValue = float_view[i / 4];
+
+        let t = 0.0;
+        const defined = !(
+            Number.isNaN(propertyValue) ||
+            propertyValue === undefined ||
+            propertyValue === null ||
+            propertyValue === 0
+        );
+        if (defined) {
+            t = propertyValue;
+            t = (t - colorMapRangeMin) / (colorMapRangeMax - colorMapRangeMin);
+            t = Math.max(0.0, t);
+            t = Math.min(1.0, t);
+        }
+
+        const color = colors[Math.floor(t * 255.0)];
+        data[i + 0] = color[0];
+        data[i + 1] = color[1];
+        data[i + 2] = color[2];
+        data[i + 3] = defined ? 255 : 0; // Undefined values is set invisible.
+    }
+
+    return resolved_image;
+}
+
+export interface MapLayerProps<D> extends ExtendedLayerProps<D> {
+    // Url to png image representing the height mesh.
+    mesh: string;
+
+    // Bounding box of the terrain mesh, [minX, minY, maxX, maxY] in world coordinates
+    bounds: [number, number, number, number];
+
+    // Mesh error in meters. The output mesh is in higher resolution (more vertices) if the error is smaller.
+    meshMaxError: number;
+
+    // Url to png image for map properties. (ex, poro or perm values as a texture)
+    propertyTexture: string;
+
+    // Rotates around 'bounds' upper left corner counterclockwise in degrees.
+    rotDeg: number;
+
+    // Contourlines reference point and interval.
+    contours: [number, number];
+
+    // Min and max property values. If not given these will be calcultated automatically from the data.
+    valueRange: [number, number];
+
+    // Name of color map. E.g "PORO"
+    colorMapName: string;
+
+    // Use color map in this range.
+    colorMapRange: [number, number];
+
+    // If true readout will be z value (depth). Otherwise it is the texture property value.
+    isReadoutDepth: boolean;
+
+    // Will calculate normals and enable phong shading.
+    enableSmoothShading: boolean;
+}
+
+export default class MapLayer extends CompositeLayer<
+    unknown,
+    MapLayerProps<unknown>
+> {
+    renderLayers(): [PrivateMeshLayer] {
+        let mesh = load(this.props.mesh, TerrainLoader, {
```

Comment:
The mesh needs the same encoding as the texture. The reason is that we don't know in advance which map is depth and which is another property.